docs/aws/audit/dynamodbmonitoring/rules/sagemaker_notebook_instance_vpc_remediation.mdx

```diff
@@ -1,6 +1,263 @@
+       PolicyDocument=json.dumps(policy_document)
+   )
+
+   print(response)
+   ```
+
+By following these steps, you can ensure that all Amazon SageMaker Notebook Instances are created within a specified VPC, thereby preventing misconfigurations related to VPC settings.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the Amazon SageMaker dashboard. Click on "Notebook instances" in the left-hand navigation pane.
+3. In the list of notebook instances, select the instance you want to check.
+4. In the details pane, look for the "VPC" field. If the field is empty or the VPC ID is not the expected one, then the Amazon SageMaker Notebook Instance is not correctly configured in the VPC.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Once you have AWS CLI installed and configured, you can start using it to interact with AWS services.
+
+2. To check Amazon SageMaker Notebook Instance in VPC, you can use the `describe-notebook-instance` command. This command returns information about a notebook instance. Here is an example of how to use this command:
+
+   ```
+   aws sagemaker describe-notebook-instance --notebook-instance-name my-notebook-instance
+   ```
+
+   Replace `my-notebook-instance` with the name of your notebook instance. This command returns a JSON object that contains information about the notebook instance.
+
+3. In the returned JSON object, look for the `DirectInternetAccess` field. If the value of this field is `Enabled`, it means that the notebook instance is not in a VPC. If the value is `Disabled`, it means that the notebook instance is in a VPC.
+
+4. To check if the notebook instance is in a DynamoDB, you can use the `describe-table` command. This command returns information about a table in DynamoDB. Here is an example of how to use this command:
+
+   ```
+   aws dynamodb describe-table --table-name my-table
+   ```
+
+   Replace `my-table` with the name of your table. This command returns a JSON object that contains information about the table. In the returned JSON object, look for the `Table` field. If the notebook instance is in the table, it should be listed in this field.
+</Accordion>
+
+<Accordion title='Using Python'>
+To check if Amazon SageMaker Notebook Instances are in a VPC, you can use the AWS SDK for Python (Boto3). Here are the steps:
+
+1. **Setup AWS SDK for Python (Boto3):**
+   First, you need to set up Boto3. Install it using pip:
+   ```
+   pip install boto3
+   ```
+   Then, configure your AWS credentials. You can do this by setting the following environment variables:
+   ```
+   AWS_ACCESS_KEY_ID = 'your_access_key'
+   AWS_SECRET_ACCESS_KEY = 'your_secret_key'
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>